packages/repository/src/mixins/repository.mixin.ts

```diff
@@ -289,12 +286,45 @@ export function RepositoryMixin<T extends MixinTarget<Application>>(
 
         if (operation in ds && typeof ds[operation] === 'function') {
           debug('Migrating dataSource %s', b.key);
-          await ds[operation](options.models);
+
+          const models = this.filterDataSourceModels(options.models, ds);
+          await ds[operation](models);
+          // if (!options.models || options.models.length === 0) {
+          //   // if options.models is an empty array, proceed normally
+          //   await ds[operation](options.models);
+          // } else {
+          //   // if options.models is not empty, check if model is
+          //   // connected to the datasource
+          //   const models = [];
+          //   for (const model of options.models) {
+          //     // todo : add "definitions" attribute in Datasource definition
+          //     // in "loopback-datasource-juggler"
+          //     if (Object.keys(ds['definitions']).includes(model)) {
+          //       models.push(model);
+          //     }
+          //   }
+          //   await ds[operation](models);
+          // }
         } else {
           debug('Skipping migration of dataSource %s', b.key);
         }
       }
     }
+
+    /**
+     * Check if a model is part of datasource
+     *
+     * Used by migrate schema
```

Comment:
I find this comment as a bit misleading. When you say "check if ...", I would expect the function to return `true` or `false` based on the check outcome.

I also feel it's not necessary to explain where is this function used - this fact can change in the future, plus it's easy to find using "find all references" or a string-based search.

Proposed text:

```suggestion
     * Filter the list of models to keep only the models attached to the given datasource.
```

Reply:
thanks for the description i was stuck ^^